Dockerfile

```
@@ -1,6 +1,26 @@
+#
+# BUILDER
+#
+FROM docker.io/library/golang:1.21.1 as builder
+
+WORKDIR /buildroot
+COPY ./ ./
+
+RUN go build -o controller-manager cmd/controller-manager/*.go
+
+
+#
+# FINAL IMAGE
+#
 FROM gcr.io/distroless/static:latest
+
 LABEL maintainers="Kubernetes Authors"
 LABEL description="COSI Controller"
 
-COPY ./bin/controller-manager controller-manager
+LABEL org.opencontainers.image.title="COSI Controller"
+LABEL org.opencontainers.image.description="Container Object Storage Interface (COSI) Controller"
+LABEL org.opencontainers.image.source="https://github.com/kubernetes-sigs/container-object-storage-interface-controller"
+LABEL org.opencontainers.image.licenses="APACHE-2.0"
+
+COPY --from=builder /buildroot/controller-manager .
```

```suggestion
COPY --from=builder /buildroot/build/controller-manager .
```

Necessary because of my previous comment.